Source/8_UI/FlowCoordinator/Components/ReplayLaunchView/BeatmapReplayLaunchPanel/ReplayDetail/DownloadBeatmapPanel.cs

```diff
@@ -118,7 +118,7 @@ private void UpdateVisibility(State state) {
                     _infoPanelObject.SetActive(true);
                     break;
                 case State.ReadyToDownload:
-                    _mapPreviewImage.SetImage(_mapDetail!.versions![0].coverURL);
+                    await _mapPreviewImage.SetImageAsync(_mapDetail!.versions![0].coverURL);
```

Comment:
I guess we dont need to wait for the image to load as buttons will be stuck not interactable. Just leave it and let it do the job.